src/main/webapp/app/shared/user-settings/notification-settings/notification-settings.service.ts

```diff
@@ -0,0 +1,91 @@
+     * @param notification which should be checked if it is activated in the notification settings of the current user
+     * @param notificationTitleActivationMap hold the information of the saved notification settings and their status
+     * @return true if this notification (title) is activated in the settings, else return false
+     */
+    public isNotificationAllowedBySettings(notification: Notification, notificationTitleActivationMap: Map<string, boolean>): boolean {
+        if (notification instanceof GroupNotification || notification.notificationType === NotificationType.GROUP || notification.notificationType === NotificationType.SINGLE) {
+            if (notification.title) {
+                return notificationTitleActivationMap.get(notification.title) ?? true;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * This is the place where the mapping between NotificationOptionCores and NotificationTypes happens on the client side
+     * Each NotificationOptionCore can be based on multiple different NotificationTypes
+     * @param notificationOptionCore which corresponding NotificationTypes should be found
+     * @return the corresponding NotificationType(s)
+     */
+    private static findCorrespondingNotificationTypesForNotificationOptionCore(notificationOptionCore: NotificationOptionCore): string[] {
+        switch (notificationOptionCore.optionSpecifier) {
+            case OptionSpecifier.NOTIFICATION__EXERCISE_NOTIFICATION__EXERCISE_CREATED_OR_STARTED: {
+                return [EXERCISE_CREATED_TITLE];
+            }
+            case OptionSpecifier.NOTIFICATION__EXERCISE_NOTIFICATION__EXERCISE_OPEN_FOR_PRACTICE: {
+                return [EXERCISE_PRACTICE_TITLE];
+            }
+            case OptionSpecifier.NOTIFICATION__EXERCISE_NOTIFICATION__NEW_POST_EXERCISES: {
+                return [NEW_POST_FOR_EXERCISE_TITLE];
+            }
+            case OptionSpecifier.NOTIFICATION__EXERCISE_NOTIFICATION__NEW_ANSWER_POST_EXERCISES: {
+                return [NEW_ANSWER_POST_FOR_EXERCISE_TITLE];
+            }
+            case OptionSpecifier.NOTIFICATION__LECTURE_NOTIFICATION__ATTACHMENT_CHANGES: {
+                return [ATTACHMENT_CHANGE_TITLE];
+            }
+            case OptionSpecifier.NOTIFICATION__LECTURE_NOTIFICATION__NEW_POST_FOR_LECTURE: {
+                return [NEW_POST_FOR_LECTURE_TITLE];
+            }
+            case OptionSpecifier.NOTIFICATION__LECTURE_NOTIFICATION__NEW_ANSWER_POST_FOR_LECTURE: {
+                return [NEW_ANSWER_POST_FOR_LECTURE_TITLE];
+            }
+            case OptionSpecifier.NOTIFICATION__INSTRUCTOR_EXCLUSIVE_NOTIFICATIONS__COURSE_AND_EXAM_ARCHIVING_STARTED: {
+                return [EXAM_ARCHIVE_STARTED_TITLE, COURSE_ARCHIVE_STARTED_TITLE];
+            }
+            default: {
+                return [];
+            }
+        }
+    }
```

Comment:
This file also seems to have a high complexity. It seems similiar to the java counterpart you had. 
Maybe you could also refactor this with much less code and complexity. I would suggest something like this: (pseudocode, not tested):

```
const mappings: Map<OptionSpecifier, string[]> = new Map([
            [OptionSpecifier.NOTIFICATION__EXERCISE_NOTIFICATION__EXERCISE_CREATED_OR_STARTED, [EXERCISE_CREATED_TITLE]],
            [OptionSpecifier.NOTIFICATION__EXERCISE_NOTIFICATION__EXERCISE_OPEN_FOR_PRACTICE, [EXERCISE_PRACTICE_TITLE]],
            [OptionSpecifier.NOTIFICATION__EXERCISE_NOTIFICATION__NEW_POST_EXERCISES, [NEW_POST_FOR_EXERCISE_TITLE]],
            [OptionSpecifier.NOTIFICATION__EXERCISE_NOTIFICATION__NEW_ANSWER_POST_EXERCISES, [NEW_ANSWER_POST_FOR_EXERCISE_TITLE]],
            [OptionSpecifier.NOTIFICATION__LECTURE_NOTIFICATION__ATTACHMENT_CHANGES, [ATTACHMENT_CHANGE_TITLE]],
            [OptionSpecifier.NOTIFICATION__LECTURE_NOTIFICATION__NEW_POST_FOR_LECTURE, [NEW_POST_FOR_LECTURE_TITLE]],
            [OptionSpecifier.NOTIFICATION__LECTURE_NOTIFICATION__NEW_ANSWER_POST_FOR_LECTURE, [NEW_ANSWER_POST_FOR_LECTURE_TITLE]],
            [OptionSpecifier.NOTIFICATION__INSTRUCTOR_EXCLUSIVE_NOTIFICATIONS__COURSE_AND_EXAM_ARCHIVING_STARTED, [EXAM_ARCHIVE_STARTED_TITLE, COURSE_ARCHIVE_STARTED_TITLE]]
        ]);
        const foundOption = mappings.get(notificationOptionCore.optionSpecifier);
        return foundOption ? foundOption : [];
```
I think this is much neater, and does the same in less lines.